lib/DXIL/DxilOperations.cpp

```diff
@@ -3404,11 +3404,13 @@ void OP::FixOverloadNames() {
       CallInst *CI = cast<CallInst>(*F.user_begin());
       DXIL::OpCode opCode = OP::GetDxilOpFuncCallInst(CI);
       llvm::Type *Ty = OP::GetOverloadType(opCode, &F);
-      if (isa<StructType>(Ty) || isa<PointerType>(Ty)) {
-        std::string funcName;
-        if (OP::ConstructOverloadName(Ty, opCode, funcName)
-                .compare(F.getName()) != 0) {
-          F.setName(funcName);
+      if (OP::IsOverloadLegal(opCode, Ty)) {
```

Comment:
nit: We can eliminate at least one or two levels of indentation using `continue` expressions.

(see: https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code)

Reply:
Updated.